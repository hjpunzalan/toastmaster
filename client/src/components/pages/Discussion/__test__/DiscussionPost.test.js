
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import { createBrowserHistory } from "history";
import DiscussionPostRouter, {DiscussionPost} from "../DiscussionPost";



const history = createBrowserHistory();
jest.spyOn(history, "push");

const initialProps = {
	img: "imgurl",
	count: 5,
	id: "test",
	title:"title",
	date:"2019-09-21T17:45:46.965Z",
	text: "test post",
	firstName: "firstname",
	lastName:"lastName"
}

const setup = (props = {}) => {
	const setupProps = { ...initialProps, ...props };
	return shallow(<DiscussionPost {...setupProps} />);
};

test('render Discussion post ', () => {
    const wrapper = setup();
    expect(wrapper.find(".Discussion__post").length).toBe(1)
})


test("Snapshot renders correctly", () => {
	const tree = renderer
		.create(<BrowserRouter><DiscussionPostRouter {...initialProps} /></BrowserRouter>)
		.toJSON();
	expect(tree).toMatchSnapshot()

})



