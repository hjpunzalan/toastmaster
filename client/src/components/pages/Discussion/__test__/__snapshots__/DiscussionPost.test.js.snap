// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders correctly 1`] = `
<a
  className="Discussion__post"
  href="/discussion/post/test"
  onClick={[Function]}
>
  <div
    className="Discussion__postUser"
  >
    <img
      alt="user-logo"
      className="Discussion__postUser-img"
      src="imgurl"
    />
    <span
      className="Discussion__postUser-name"
    >
      <span>
        firstname
        Â 
      </span>
      <span>
        lastName
      </span>
    </span>
  </div>
  <div
    className="Discussion__postBody"
  >
    <div
      className="Discussion__postBody-header"
    >
      <h3
        className="Discussion__postBody-title"
      >
         
        title
      </h3>
      <span
        className="Discussion__postBody-chat"
      >
        5
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M425.9 170.4H204.3c-21 0-38.1 17.1-38.1 38.1v154.3c0 21 17.1 38 38.1 38h126.8c2.8 0 5.6 1.2 7.6 3.2l63 58.1c3.5 3.4 9.3 2 9.3-2.9v-50.6c0-6 3.8-7.9 9.8-7.9h1c21 0 42.1-16.9 42.1-38V208.5c.1-21.1-17-38.1-38-38.1z"
          />
          <path
            d="M174.4 145.9h177.4V80.6c0-18-14.6-32.6-32.6-32.6H80.6C62.6 48 48 62.6 48 80.6v165.2c0 18 14.6 32.6 32.6 32.6h61.1v-99.9c.1-18 14.7-32.6 32.7-32.6z"
          />
        </svg>
      </span>
    </div>
    <p
      className="Discussion__postBody-text"
    >
      test post
    </p>
    <span
      className="Discussion__postBody-date"
    >
      <time
        dateTime={"2019-09-21T17:45:46.965Z"}
      >
        22 Sep 2019
      </time>
    </span>
  </div>
</a>
`;
